vnext/ReactUWP/Views/Image/ReactImageBrush.cpp

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include "ReactImageBrush.h"
+
+#include <sstream>
+
+#include <winrt/Windows.UI.Composition.h>
+
+#include "BorderEffect.h"
+
+namespace winrt {
+  using namespace winrt::Windows::Storage::Streams;
+  using namespace winrt::Windows::UI::Composition;
+  using namespace winrt::Windows::UI::Xaml;
+  using namespace winrt::Windows::UI::Xaml::Media;
+}
+
+namespace react {
+  namespace uwp {
+    /*static*/ winrt::com_ptr<ReactImageBrush> ReactImageBrush::Create()
+    {
+      return winrt::make_self<ReactImageBrush>();
+    }
+
+    void ReactImageBrush::OnConnected()
+    {
+      UpdateCompositionBrush();
+    }
+
+    void ReactImageBrush::OnDisconnected()
+    {
+      // Dispose of composition resources when no longer in use.
+      if (CompositionBrush())
+      {
+        CompositionBrush(nullptr);
+      }
+    }
+
+    void ReactImageBrush::ResizeMode(react::uwp::ResizeMode value)
+    {
+      if (m_resizeMode != value)
+      {
+        m_resizeMode = value;
+        UpdateCompositionBrush();
+      }
+    }
+
+    void ReactImageBrush::AvailableSize(winrt::Size const& value)
+    {
+      if (m_availableSize != value)
+      {
+        m_availableSize = value;
+        UpdateCompositionBrush();
+      }
+    }
+
+    void ReactImageBrush::Source(winrt::LoadedImageSurface const& value)
+    {
+      if (m_loadedImageSurface != value)
+      {
+        m_loadedImageSurface = value;
+        UpdateCompositionBrush();
+      }
+    }
+
+    void ReactImageBrush::UpdateCompositionBrush()
+    {
+      if (m_loadedImageSurface)
+      {
+        winrt::CompositionSurfaceBrush surfaceBrush{ GetOrCreateSurfaceBrush() };
+        surfaceBrush.Stretch(ResizeModeToStretch());
+
+        auto compositionBrush{ surfaceBrush.as<winrt::CompositionBrush>() };
+        if (ResizeMode() == ResizeMode::Repeat)
+        {
+          compositionBrush = GetOrCreateEffectBrush(surfaceBrush);
+        }
+
+        // Only switch CompositionBrush if we are switching to/from ResizeMode::Repeat
+        if (ShouldSwitchCompositionBrush())
```

Comment:
Should this check be made before creating surfaceBrush and compositionBrush?

Reply:
It was a bad name. The code above determines the correct brush we need to use. This should just check to see if it is different than what we're currently using and set it if that's the case. 

---
In reply to: [294561221](https://github.com/microsoft/react-native-windows/pull/2581#discussion_r294561221) [](ancestors = 294561221)